diff --git a/node_modules/@bacons/apple-targets/build/with-xcode-changes.js b/node_modules/@bacons/apple-targets/build/with-xcode-changes.js
index 15a7aa2..49d2ff1 100644
--- a/node_modules/@bacons/apple-targets/build/with-xcode-changes.js
+++ b/node_modules/@bacons/apple-targets/build/with-xcode-changes.js
@@ -184,20 +184,19 @@ async function applyXcodeChanges(config, project, props) {
         }
     }
     if (targetToUpdate) {
+        const existingBuildConfigurationList = targetToUpdate.props.buildConfigurationList;
         // Remove existing build phases
-        targetToUpdate.props.buildConfigurationList.props.buildConfigurations.forEach((config) => {
+        existingBuildConfigurationList === null || existingBuildConfigurationList === void 0 ? void 0 : existingBuildConfigurationList.props.buildConfigurations.forEach((config) => {
             config.getReferrers().forEach((ref) => {
                 ref.removeReference(config.uuid);
             });
             config.removeFromProject();
         });
         // Remove existing build configuration list
-        targetToUpdate.props.buildConfigurationList
-            .getReferrers()
-            .forEach((ref) => {
-            ref.removeReference(targetToUpdate.props.buildConfigurationList.uuid);
+        existingBuildConfigurationList === null || existingBuildConfigurationList === void 0 ? void 0 : existingBuildConfigurationList.getReferrers().forEach((ref) => {
+            ref.removeReference(existingBuildConfigurationList.uuid);
         });
-        targetToUpdate.props.buildConfigurationList.removeFromProject();
+        existingBuildConfigurationList === null || existingBuildConfigurationList === void 0 ? void 0 : existingBuildConfigurationList.removeFromProject();
         // Create new build phases
         targetToUpdate.props.buildConfigurationList =
             (0, configuration_list_1.createConfigurationListForType)(project, props);
diff --git a/node_modules/@bacons/apple-targets/ios/ExtensionStorageModule.swift b/node_modules/@bacons/apple-targets/ios/ExtensionStorageModule.swift
index 56a949b..e423ec8 100644
--- a/node_modules/@bacons/apple-targets/ios/ExtensionStorageModule.swift
+++ b/node_modules/@bacons/apple-targets/ios/ExtensionStorageModule.swift
@@ -1,3 +1,4 @@
+import Foundation
 import ExpoModulesCore
 import WidgetKit
 
@@ -6,14 +7,18 @@ public class ExtensionStorageModule: Module {
         Name("ExtensionStorage")
         
         Function("remove") { (forKey: String, suiteName: String?) in
-            UserDefaults(suiteName: suiteName)?.removeObject(forKey: forKey)
+            let userDefaults = UserDefaults(suiteName: suiteName)
+            userDefaults?.removeObject(forKey: forKey)
+            userDefaults?.synchronize()
         }
         
         Function("reloadWidget") { (timeline: String?) in
-            if let timeline = timeline {
-                WidgetCenter.shared.reloadTimelines(ofKind: timeline)
-            } else {
-                WidgetCenter.shared.reloadAllTimelines()
+            DispatchQueue.main.async {
+                if let timeline = timeline {
+                    WidgetCenter.shared.reloadTimelines(ofKind: timeline)
+                } else {
+                    WidgetCenter.shared.reloadAllTimelines()
+                }
             }
         }
         
@@ -31,7 +36,9 @@ public class ExtensionStorageModule: Module {
             // Convert the incoming array of dictionaries directly to JSON data
             do {
                 let jsonData = try JSONSerialization.data(withJSONObject: data, options: [])
-                UserDefaults(suiteName: suiteName)?.set(jsonData, forKey: forKey)
+                let userDefaults = UserDefaults(suiteName: suiteName)
+                userDefaults?.set(jsonData, forKey: forKey)
+                userDefaults?.synchronize()
                 return true
             } catch {
                 // If encoding fails for some reason, return false
@@ -42,7 +49,9 @@ public class ExtensionStorageModule: Module {
         Function("setObject") { (forKey: String, data: [String: Any], suiteName: String?) -> Bool in
             do {
                 let jsonData = try JSONSerialization.data(withJSONObject: data, options: [])
-                UserDefaults(suiteName: suiteName)?.set(jsonData, forKey: forKey)
+                let userDefaults = UserDefaults(suiteName: suiteName)
+                userDefaults?.set(jsonData, forKey: forKey)
+                userDefaults?.synchronize()
                 return true
             } catch {
                 // If encoding fails for some reason, return false
@@ -53,11 +62,13 @@ public class ExtensionStorageModule: Module {
         Function("setInt") { (key: String, value: Int, group: String?) in
             let userDefaults = UserDefaults(suiteName: group)
             userDefaults?.set(value, forKey: key)
+            userDefaults?.synchronize()
         }
         
         Function("setString") { (key: String, value: String, group: String?) in
             let userDefaults = UserDefaults(suiteName: group)
             userDefaults?.set(value, forKey: key)
+            userDefaults?.synchronize()
         }        
 
         Function("get") { (key: String, group: String?) -> String? in
